arch/x86/kvm/vmx/pkvm_host.c

```diff
@@ -5,6 +5,30 @@
 #include "x86_ops.h"
 #include "vmx.h"
 
+static void *kvm_host_va(phys_addr_t phys)
+{
+	return __va(phys);
+}
+
+static void host_free_pkvm_page_range(struct pkvm_page_range range)
+{
+	void *vaddr = __va(range.addr);
+	u64 nr_pages = range.nr_pages;
+
+	if (WARN_ON_ONCE(!nr_pages))
+		return;
+
+	if (nr_pages > 1)
+		free_pages_exact(vaddr, nr_pages << PAGE_SHIFT);
+	else
+		free_page((unsigned long)vaddr);
+}
+
+static void host_free_pkvm_memcache(struct pkvm_memcache *mc)
```

Comment:
This function is gonna be reused by pvMMU (and maybe even by pvIOMMU, as it turns out), i.e. outside this file. So maybe let's proactively move it to arch/x86/kvm/pkvm.c and maybe change its name to a more suitable name for an API function?

In v6.12 this function is named just free_pkvm_memcache(), and the underlying helper it uses is named __free_pkvm_memcache(). So maybe let's keep the same naming? (It is also consistent with the naming in pKVM-ARM.)

@jaszczyk-grzegorz @vineethrp for visibility.

Reply:
Yeah, `kvm_free_pkvm_memcache()` is a good name. I will rename it and move it as well as the other related functions to the pkvm.c